cpp/include/cuspatial/detail/utility/device_atomics.cuh

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <algorithm>
+#include <type_traits>
+
+namespace {
+
+/**
+ * @internal
+ * This helper is required to avoid ambiguous lookup of std::min. There are
```

Comment:
Should we really be using `std::min/std::max` at all? CUDA provides `min()` and `max()` overloads for float and double which map directly to hardware instructions.  Moreover, calling `std::min() / `std::max()` only  compiles successfully if you pass `--expt-relaxed-constexpr` to NVCC on the command line as this example shows: https://godbolt.org/z/8b85bx6e1

And this example (with that flag passed), shows that `std::min` results in  different code and more instructions (for the `float` case, which uses `FMNMX`).

https://godbolt.org/z/e6xcnTK99

In the `double` case, the code is different as well, but doesn't map directly to a `FMNMX` instruction: https://godbolt.org/z/afM7vP9od

I think we should stick to the NVIDIA math libraries in this case.  libcudac++ should probably add min and max that map to CUDA min and max in device code for consistency.

Reply:
I agree, just calling `min`/`max` is probably what we should do here.

Reply:
Done.